### DISCUSSION TITLE
Request for help with using !important in HTML tags for custom components

### DISCUSSION THREAD

**Discussion by memiljamel** [tailwindlabs/tailwindcss#12296] [Help] [answered]
I'm trying to apply the !important flag to a custom class within my HTML tags, and I'm facing some challenges. I want to ensure that the styles are marked as important directly within the HTML tag, but it doesn't seem to work as expected.

> I've been looking for a similar problem, only in my case I don't want to put !important directly in the component's custom class.

I've tried using class components like this:
`<div class="body-2 !subtitle-1"></div>`

and this is my custom components class (app.css):
```
@tailwind base;
@tailwind components;
@tailwind utilities;

@layer components {
 
  .subtitle-1 {
    @apply font-roboto font-normal text-base tracking-[0.15px];
  }

  .body-2 {
    @apply font-roboto font-normal text-sm tracking-[0.25px];
  }
}
```

results:
[https://i.imgur.com/4HwbPkL.png](https://i.imgur.com/4HwbPkL.png)

Is there a way to make !important a custom component work directly in the HTML tag? I would appreciate any insight or guidance on this matter. Thank you for your help!

Please let me know if you need further context or information.

**Comment by adydetra** [comment]
Remove `font-roboto`

Error preview

![image](https://github.com/tailwindlabs/tailwindcss/assets/108399634/6aa30409-38f7-40dd-a6ec-cb5c4a5b5f3a)

---

Use "!" here, like this

```css
@tailwind base;
@tailwind components;
@tailwind utilities;

@layer components {
  .subtitle-1 {
    @apply !text-blue-500 font-normal !text-7xl tracking-[0.15px] !border-red-500;
  }

  .body-2 {
    @apply text-red-500 font-normal text-sm tracking-[0.25px] border-b border-blue-500;
  }
}
```

Result

![image](https://github.com/tailwindlabs/tailwindcss/assets/108399634/43d1724b-fe5c-420d-a331-c2b07c2ce0af)

**Reply by memiljamel** [reply]
Of course an error occurred, because you didn't use a custom font. Here my problem is that I can't use the directive important '!' for my custom component in html directly, not in component (app.css).

**Reply by adydetra** [reply]
As far as I know, that's not possible, I've never seen a tailwindcss developer use important in his custom components. or maybe I don't know 👀.

**Comment by wongjn** [comment, marked as answer]
You could consider not using `@apply`. The important modifier seems to work with regular CSS declarations: https://play.tailwindcss.com/fxZN9e4aeR?file=css
